crates/bevy_color/src/hsla.rs

```diff
@@ -166,6 +168,35 @@ impl Luminance for Hsla {
     }
 }
 
+impl ClampColor for Hsla {
+    fn clamped(&self) -> Self {
```

Comment:
Rather than repeating the formula twice for each color space, you can provide a default implementation of clamp in terms of clamped or vice versa.

```
fn clamp(&self) {
    *self = self.clamped();
}
```